src/librustc/traits/select.rs

```diff
@@ -1325,17 +1325,21 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         (result, dep_node)
     }
 
-    // Treat negative impls as unimplemented
-    fn filter_negative_impls(
+    // Treat negative impls as unimplemented, and reservation impls as Ok(None)
```

Comment:
instead of "as `Ok(None)`, perhaps:

Treat negative impls as unimplemented, and reservation impls as ambiguity.